src/components/header.js

```diff
@@ -32,29 +34,90 @@ function Header({ slug }) {
+  if (translationFileLength !== constFileLength)
+    console.warn(
+      "Different number of dropdowns in /src/constants/link.js and dropdown labels in /src/locales\n",
+      `${constFileLength} dropdown${
+        constFileLength > 1 ? "s" : ""
+      } in /src/constants/link.js\n`,
+      `${translationFileLength} dropdown${
+        translationFileLength > 1 ? "s" : ""
+      } in /src/locales`
+    );
+  dropdownLinks.map((_, i) => {
+    if (i < length) {
+      const navDropdownLinks = dropdownLinks[i];
+      const navDropdownLinksLength = navDropdownLinks.length;
+      const navDropdownLinkLabels = t("header.nav.dropdowns")[i]["simpleLinks"];
+      const navDropdownLinkLabelsLength = navDropdownLinkLabels.length;
+      const navDropdownLength =
+        navDropdownLinksLength > navDropdownLinkLabelsLength
+          ? navDropdownLinkLabelsLength
+          : navDropdownLinksLength; // take shorter length in case there is a missing link in `../constants/links.js` (constants file) or label in  `../locales/` (translation files)
+      if (navDropdownLinksLength !== navDropdownLinkLabelsLength)
+        console.warn(
+          `Different number of links in /src/constants/link.js and link labels in /src/locales for dropdown ${
+            i + 1
+          }\n`,
+          "Links: ",
+          navDropdownLinks,
+          "\n",
+          "Link labels: ",
+          navDropdownLinkLabels,
+          "\n"
+        );
+      dropdowns.push(
+        navDropdownLinks.map((element, i) => {
+          return (
+            i < navDropdownLength && (
+              <Link to={element} key={i}>
+                {navDropdownLinkLabels[i]}
+              </Link>
+            )
+          );
+        })
+      );
+    } else dropdowns.push([]);
+  });
 
+  /* dynamically creating parent links */
+  const parentLinks = links.parent;
+  const parentLinksLength = parentLinks.length;
+  const parentLinksLabels = t("header.nav.parentLinks");
+  const parentLinksLabelsLength = parentLinksLabels.length;
+  const parentLength =
+    parentLinksLength > parentLinksLabelsLength
+      ? parentLinksLabelsLength
+      : parentLinksLength; // take shorter length in case there is a missing link in parentLinks or missing label in translation file
+  if (parentLinksLength !== parentLinksLabelsLength)
+    console.warn(
+      "Different number of parent links in /src/constants/link.js and parent labels in /src/locales\n",
+      "Links: ",
+      parentLinks,
+      "\n",
+      "Labels: ",
+      parentLinksLabels,
+      "\n"
+    );
+  const parentLinkItems = parentLinks.map((element, i) => {
+    return (
+      i < parentLength && (
+        <Link to={element} variant="nav" key={i}>
+          <span>{parentLinksLabels[i]}</span>
+        </Link>
+      )
+    );
+  });
   const navBarItems = [
```

Comment:
`navBarItems` can be created dynamically too by looping through `dropdowns`. This would be useful if the user wanted more or less than 2 nav dropdowns

Reply:
This will be handled in another PR.